components/supervisor/pkg/supervisor/top_test.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package supervisor
+
+import (
+	"context"
+	"testing"
+)
+
+func TestTopService(t *testing.T) {
+	ctx := context.Background()
+
+	topService := &TopService{}
+	topService.Observe(ctx)
+
+	if topService.data == nil {
```

Comment:
maybe add a ready channel to synchronise on first successful request

Reply:
@akosyakov do you mean something the client will have to check to see if data is ready? 🤔

Reply:
just for tests use internal channel in the same package
```go
<-topService.ready // returns on first successful request
if topService.data == nil {
...
```

Reply:
Updated 🙏